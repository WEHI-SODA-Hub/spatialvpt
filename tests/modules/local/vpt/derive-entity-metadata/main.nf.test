nextflow_process {

    name "Test Process DERIVE_ENTITY_METADATA"
    script "modules/local/vpt/derive-entity-metadata/main.nf"
    process "DERIVE_ENTITY_METADATA"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/tests/data/cellpose_micron_space.parquet', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
