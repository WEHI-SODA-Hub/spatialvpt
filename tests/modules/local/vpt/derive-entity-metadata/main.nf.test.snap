{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cell_metadata_resegmented.csv:md5,d09b988486aca2950e3aeb151e3a8765"
                    ]
                ],
                "1": [
                    "versions.yml:md5,239295f21e6d2a3c78b5784e7f573d05"
                ],
                "entity_metadata": [
                    [
                        {
                            "id": "small_set"
                        },
                        "cell_metadata_resegmented.csv:md5,d09b988486aca2950e3aeb151e3a8765"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,239295f21e6d2a3c78b5784e7f573d05"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-06T09:58:17.34148161"
    }
}